pkg/flowaggregator/flowaggregator.go

```diff
@@ -557,6 +557,10 @@ func (fa *flowAggregator) GetRecordMetrics() querier.Metrics {
 		NumRecordsReceived: fa.collectingProcess.GetNumRecordsReceived(),
 		NumFlows:           fa.aggregationProcess.GetNumFlows(),
 		NumConnToCollector: fa.collectingProcess.GetNumConnToCollector(),
+		ClickHouseExporter: fa.clickHouseExporter != nil && fa.clickHouseExporter.IsRunning(),
+		S3Exporter:         fa.s3Exporter != nil && fa.s3Exporter.IsRunning(),
+		LogExporter:        fa.logExporter != nil && fa.logExporter.IsRunning(),
+		IPFIXExporter:      fa.ipfixExporter != nil && fa.ipfixExporter.IsRunning(),
```

Comment:
For the e2e test, I just want to know whether a specific exporter has called its Start() func. Currently, I think we can only check the connection for a ClickHouseExporter before running Start(). The connection status for the other 3 exporters will only be revealed when we send the record to them.

Or should I set the isRunning flag after the record can be successfully send to a specific exporter?

Reply:
My preference would be to avoid tying this to the `Start` method altogether.
Could we do the following:
* replace `IsExporterXRunning` with `WithExporterX`, and set it to true iff `fa.ExporterX != nil`
* use a combination of `WithExporterX` and of the existing `NumRecordsExported` in tests

Would that work?